Benchmarks/DualityBenchmarks/CloneProviderBenchmarks.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using BenchmarkDotNet.Attributes;
+using Duality.Cloning;
+using Duality.Tests.Serialization;
+
+namespace DualityBenchmarks
+{
+	[MemoryDiagnoser]
+	public class CloneProviderBenchmarks
+	{
+		private Random rnd;
+		private TestObject[] results;
+		private TestObject data;
+
+		[GlobalSetup]
+		public void Setup()
+		{
+			this.rnd = new Random(0);
+			this.results = new TestObject[200];
+			this.data = new TestObject(this.rnd, 5);
+		}
+
+		[Benchmark]
+		public void CloneTestObjectGraph()
+		{
+			for (int i = 0; i < this.results.Length; i++)
+			{
+				this.results[i] = this.data.DeepClone();
+			}
+		}
+
+		[Benchmark(Baseline = true)]
+		public void CreateWithoutClone()
+		{
+			for (int i = 0; i < this.results.Length; i++)
+			{
+				this.results[i] = new TestObject(this.rnd, 5);
```

Comment:
Some thing I noticed that here we are actually measuring the performance of the TestObject constructor (which is quite inefficient due to alot of unneeded allocations) which might not be the best baseline case but fixing that is out of the scope of this PR.

Reply:
Good catch 👍 I was surprised by that result as well. 

(Agree that this is out of scope for this PR though.)

Reply:
Yeah, benchmarkdotnet is just way more precise than anything you could probably hand write in a decent amount of time. Combined with the detailed allocation data that just makes it easier to spot these kind of things which is a good thing.